domain/life/life.go

```diff
@@ -12,3 +16,29 @@ const (
 	Dying
 	Dead
 )
+
+// ToCoreLife converts a life value to a core life value.
+func (l *Life) ToCoreLife() corelife.Value {
+	switch *l {
+	case Alive:
+		return corelife.Alive
+	case Dying:
+		return corelife.Dying
+	case Dead:
+		return corelife.Dead
+	}
+	panic("unknown life value")
```

Comment:
Panics are not an option here. This will effectively take down a controller. I don't think we need this package.

Reply:
Taking out the life package needs to be done in a separate PR as it will grow this one even bigger. Blockdevice domain uses that internally, and without that package we need to query the life table wherever we produce a (core) life value for life comparisons, as the core life values are strings. Currently we keep the `life_id` in the machine table as an integer against the `life` table.

```sql
CREATE TABLE life (
    id INT PRIMARY KEY,
    value TEXT NOT NULL
);

INSERT INTO life VALUES
(0, 'alive'),
(1, 'dying'),
(2, 'dead');
```

so to do something like `machineLifeID == life.Dead` where we get the machineLifeID from the machine table, we need to query the `life` table and do the comparison like `machineLifeValue == corelife.Dead`.

I will remove the panics though.